geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientProxy.java

```diff
@@ -1673,13 +1673,16 @@ MessageDispatcher createMessageDispatcher(String name) {
     return messageDispatcherFactory.create(this, name, statisticsClock);
   }
 
-  protected void startOrResumeMessageDispatcher(boolean processedMarker) {
+  protected void startOrResumeMessageDispatcher(boolean sendMarkerDirectly) {
     // Only start or resume the dispatcher if it is Primary
     if (isPrimary) {
       // Add the marker to the queue
-      if (!processedMarker) {
-        EventID eventId = new EventID(_cache.getDistributedSystem());
-        _messageDispatcher.enqueueMarker(new ClientMarkerMessageImpl(eventId));
+      EventID eventId = new EventID(_cache.getDistributedSystem());
+      ClientMarkerMessageImpl clientMarkerMessage = new ClientMarkerMessageImpl(eventId);
+      if (sendMarkerDirectly) {
+        sendMessageDirectly(clientMarkerMessage);
```

Comment:
Will this cause unforeseen consequences if Dispatcher is not yet started/resumed but send message first? The method later starts dispatcher thread.

Reply:
I can't see any bad consequence. It's just a marker message that is supposed to go to the client anyway.